tests/validator_test.go

```diff
@@ -329,4 +338,618 @@ var _ = Describe("Template validator", func() {
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+		})
+		It("test with template rules unfulfilled", func() {
+			template = TemplateWithRules()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "128M")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Should have given error failing to fulfill validations")
+		})
+		It("Test template with incorrect rules satisfied", func() {
+			template = TemplateWithIncorrectRules()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "128M")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).To(HaveOccurred(), "Should have given error failing to fulfill validations")
+		})
+		It("Test template with incorrect rules unfulfilled", func() {
+			template = TemplateWithIncorrectRules()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "32M")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			Expect(apiClient.Create(ctx, vm)).To(HaveOccurred(), "Should have given error failing to fulfill validations")
+		})
+		It("[test_id:2959] Start a VM with memory restrictions violation that succeeds with a warning", func() {
+			template = TemplateWithIncorrectRulesJustWarning()
+			template.Namespace = strategy.GetNamespace()
+			Expect(apiClient.Create(ctx, &template)).ToNot(HaveOccurred(), "Failed to create template: %s", template.Name)
+
+			vmi = addDomainResourcesToVMI(vmi, 0, "q35", "1G")
+			vm = NewVirtualMachine(vmi, false)
+			vm.ObjectMeta.Annotations = map[string]string{
+				"vm.kubevirt.io/template":           template.Name,
+				"vm.kubevirt.io/template-namespace": template.Namespace,
+			}
+			// TODO: Investigate why this test consistently fails without Eventually
```

Comment:
Probably the same reason as the other test - race condition.

Reply:
But not sure why only on this test, consistently